cachito/workers/tasks.py

```diff
@@ -1,17 +1,48 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 import logging
+import os
+import sys
 
 from celery import Celery
+from requests import Timeout
+
+from cachito.workers.config import configure_celery, validate_celery_config
+from cachito.workers.scm import Git
+from cachito.errors import CachitoError
 
-from cachito.workers.config import configure_celery
 
 log = logging.getLogger(__name__)
 logging.basicConfig()
 app = Celery()
 configure_celery(app)
+# Only validate the configuration when Celery is directly running this module. This allows
+# importers of this module (e.g. the Flask app) to not have Celery fully configured.
+if os.path.basename(sys.argv[0]).startswith('celery'):
```

Comment:
This is pretty hackish :)
Can we use [on_init](https://docs.celeryproject.org/en/latest/reference/celery.html#celery.Celery.on_init) ?

Reply:
I think `on_init` wouldn't solve the issue, but it looks like [worker_init signal](https://docs.celeryproject.org/en/master/userguide/signals.html?highlight=signals#worker-init) might work. I'll try that.